### PR TITLE
Feature: Shacl Collection

### PR DESCRIPTION
**Pull request by WLefever-Cegeka**
First step in refactoring the Shacl Admin Controller by:
* Creating an in memory shacl shape collection that reflects the database collection
* redirecting all shacl update and retrieval operations towards this collection

Follow up PR will move away from the use of LdesConfigModel

The Validated issue will be taken up in an issue

### CODE REVIEW

File: ldes-server-port-admin-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestController.java [hunk 1]
```diff
@@ -36,22 +33,14 @@ private void initBinder(WebDataBinder binder) {
...
 	@GetMapping("/eventstreams/{collectionName}/shape")
 	public ResponseEntity<LdesConfigModel> getShape(@PathVariable String collectionName) {
-		LdesConfigModel shape = service.retrieveShape(collectionName);
+		LdesConfigModel shape = shaclCollection.retrieveShape(collectionName);
 		return ResponseEntity.ok(shape);
 	}
 
 	@PutMapping("/eventstreams/{collectionName}/shape")
-	public ResponseEntity<LdesConfigModel> putShape(@PathVariable String collectionName,
+	public void putShape(@PathVariable String collectionName,
```
Comment (Yalz) [inline ldes-server-port-admin-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestController.java:41]: no return message for PUT ?

**Review by Yalz — APPROVED** [review]

File: ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java [hunk 2]
```diff
@@ -70,78 +72,81 @@ class AdminShapeRestControllerTest {
...
-		LdesConfigModel shapeConfigModel = LdesConfigModel.createLdesConfigShape(collectionName, expectedShapeModel);
-		when(ldesConfigModelService.retrieveShape(collectionName)).thenReturn(shapeConfigModel);
-		ResultActions resultActions = mockMvc.perform(get("/admin/api/v1/eventstreams/" + collectionName + "/shape"))
-				.andDo(print())
-				.andExpect(status().isOk());
-		MvcResult result = resultActions.andReturn();
-		Model actualModel = RdfModelConverter.fromString(result.getResponse().getContentAsString(), Lang.TURTLE);
-		Assertions.assertTrue(actualModel.isIsomorphicWith(expectedShapeModel));
-	}
+	@Nested
```
Comment (Tomvbe) [inline ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java:83]: nested :heart:

File: ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java [hunk 3]
```diff
@@ -70,78 +72,81 @@ class AdminShapeRestControllerTest {
...
+			Model expectedShapeModel = readModelFromFile("shape-1.ttl");
+			when(shaclCollection.retrieveShape(collectionName))
+					.thenReturn(new LdesConfigModel(collectionName, expectedShapeModel));
+
+			ResultActions resultActions = mockMvc
+					.perform(get("/admin/api/v1/eventstreams/" + collectionName + "/shape"))
+					.andDo(print())
+					.andExpect(status().isOk());
+
+			MvcResult result = resultActions.andReturn();
```
Comment (Tomvbe) [inline ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java:97]: I have seen the below lines on a couple of places, perhaps we should consider using our own matcher. Then we could just add `.andExpect(content().string(IsIsomorphic.with(model)));` to the chain above and be done.

Comment (Tomvbe) [reply ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java:97]: Possible implementation for the matcher:

```
public class IsIsomorphic extends BaseMatcher<String> {

    private final Model model;

    IsIsomorphic(Model model) {
        this.model = model;
    }

    public static IsIsomorphic with(Model model) {
        return new IsIsomorphic(model);
    }

    @Override
    public boolean matches(Object object) {
        if (object instanceof String string) {
            Model actualModel = RdfModelConverter.fromString(string, Lang.TURTLE);
            return actualModel.isIsomorphicWith(model);
        }
        return false;
    }

    @Override
    public void describeTo(Description description) {
        description.appendText("Result should be isomorphic with provided model");
    }
}
```

File: ldes-server-port-ingestion-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingestion/rest/LdesMemberIngestionController.java [hunk 4]
```diff
@@ -35,12 +33,8 @@ public void ingestLdesMember(@RequestBody Member member,
...
 	private void validateMember(Member member, String collectionName) {
-		Model shape = shapes.get(collectionName);
-		new LdesShaclValidator(shape, appConfig.getLdesConfig(collectionName)).validate(member);
-	}
-
-	@EventListener
-	public void handleShaclChangedEvent(ShaclChangedEvent event) {
-		shapes.put(event.getCollectionName(), event.getShacl());
+		Model shape = shaclCollection.retrieveShape(collectionName).getModel();
+		if (shape != null)
```
Comment (Tomvbe) [inline ldes-server-port-ingestion-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingestion/rest/LdesMemberIngestionController.java:37]: braces

File: ldes-server-port-admin-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestController.java (hunk 1 above)
Comment (WLefever-Cegeka) [reply ldes-server-port-admin-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestController.java:41]: updated to return input (method only publishes event)

File: ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java (hunk 3 above)
Comment (WLefever-Cegeka) [reply ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java:97]: Ha, nice one, well actually in the follow-up PR we fix it by including the `isIsomorphicWith` in the `equals` method of the object :-)

File: ldes-server-port-ingestion-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingestion/rest/LdesMemberIngestionController.java (hunk 4 above)
Comment (WLefever-Cegeka) [reply ldes-server-port-ingestion-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingestion/rest/LdesMemberIngestionController.java:37]: check

**Pull request merged**